### PR TITLE
Fix alert helper library not available with component-prometheus

### PR DESCRIPTION
**Pull request by DebakelOrakel**
If moving away from rancher-monitoring to component-prometheus the alert handling helpers are not available, therefor the alertrules will not be deployed.

## Checklist

- [ ] PR contains a single logical change (to build a better changelog).
- [x] Update the documentation.
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.
- [ ] Link this PR to related issues or PRs.

### CODE REVIEW

File: component/alertrules.jsonnet [hunk 1]
```diff
@@ -6,11 +6,10 @@ local params = inv.parameters.topolvm;
...
 
-assert
-  std.member(inv.applications, 'rancher-monitoring') ||
-  std.member(inv.applications, 'openshift4-monitoring') ||
-  std.member(inv.applications, 'prometheus')
-  : 'Neither rancher-monitoring nor openshift4-monitoring nor prometheus is available';
+// assert
+//   std.member(inv.applications, 'openshift4-monitoring') ||
+//   std.member(inv.applications, 'rancher-monitoring')
+//   : 'Neither rancher-monitoring nor openshift4-monitoring is available';
```
Comment (simu) [inline component/alertrules.jsonnet]: If you want to remove stuff, remove it, don't just comment it out. However, I don't get why we want to remove this, if anything you'd just want to drop component rancher-monitoring and keep the check for either openshift4-monitoring or prometheus.

**Review by simu — COMMENTED** [review]
The copy-pasted `render_array()` should be replaced with `com.renderArray()`. On OCP4 the copy-pasted `filter_patch_rules()` should be replaced with `filterPatchRules()` function from component openshift4-monitoring on OCP4. For me it's fine to just emit the alert rules with less processing on non-OCP4 for the moment, since there's no equivalent to `filterPatchRules()` for component-prometheus.

File: component/alertrules.jsonnet (hunk 1 above)
Comment (DebakelOrakel) [reply component/alertrules.jsonnet]: Seems like an error happend there on my side, that was not the intention 😅

Comment (DebakelOrakel) [reply component/alertrules.jsonnet]: The assert rule has to go, it won't compile using component-prometheus.
Nvm 🤦‍♂️

Comment (simu) [reply component/alertrules.jsonnet]: > The assert rule has to go, it won't compile using component-prometheus.

Then we should fix it to compile with component-prometheus, or reduce the assertion to `openshift4-monitoring`.

File: .cruft.json [hunk 2]
```diff
@@ -7,7 +7,7 @@
       "name": "TopoLVM",
       "slug": "topolvm",
       "parameter_key": "topolvm",
-      "test_cases": "defaults legacy",
+      "test_cases": "defaults legacy noalertrules",
```
Comment (simu) [inline .cruft.json:10]: Looks like the new test case wasn't added through `commodore component update -t noalertrules /path/to/component`. The new test case is missing in the GitHub actions config.

**Review by simu — APPROVED** [review]

**Pull request merged**